openmetadata-clients/openmetadata-java-client/src/main/java/org/openmetadata/client/gateway/OpenMetaData.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *  Copyright 2021 Collate
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.openmetadata.client.gateway;
+
+import io.swagger.client.ApiClient;
+import io.swagger.client.api.CatalogApi;
+import io.swagger.client.model.CatalogVersion;
+import io.swagger.client.model.OpenMetadataServerConnection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.regex.Pattern;
+public class OpenMetaData {
+    private static final Logger LOG = LoggerFactory.getLogger(OpenMetaData.class);
+    private static final CatalogVersion CATALOG_VERSION_CLIENT;
+
+    static {
+        CATALOG_VERSION_CLIENT = new CatalogVersion();
+        try {
+            InputStream fileInput = OpenMetaData.class.getResourceAsStream("/catalog/VERSION");
+            Properties props = new Properties();
+            props.load(fileInput);
+            CATALOG_VERSION_CLIENT.setVersion(props.getProperty("version", "unknown"));
+            CATALOG_VERSION_CLIENT.setRevision(props.getProperty("revision", "unknown"));
+
+            String timestampAsString = props.getProperty("timestamp");
+            Long timestamp = timestampAsString != null ? Long.valueOf(timestampAsString) : null;
+            CATALOG_VERSION_CLIENT.setTimestamp(timestamp);
+        } catch (Exception ie) {
+            LOG.warn("Failed to read catalog version file");
+        }
+    }
+    private ApiClient apiClient;
+    private OpenMetadataServerConnection serverConfig;
+    private String basePath;
+    public OpenMetaData(OpenMetadataServerConnection config){
+        serverConfig = config;
+        apiClient = new ApiClient();
+        //TODO: Custom interceptor for AUTH, currently this only works for NO-AUTH, can be used for local testing
+        basePath = config.getHostPort()+ "/" + config.getApiVersion();
```

Comment:
another option is to put the /version outside of authentication and make it publicly accessible api